### PR TITLE
refactor(rgbpp|service): Add `send_rgbpp_group_txs` RPC to SDK Service

### PR DESCRIPTION
**Pull request by duanyytop**
## Changes

-  Update response of sending RGB++ group transactions
-  Add `send_rgbpp_group_txs` RPC to SDK service
- Update `generate_rgbpp_transfer_all_txs` response

### CODE REVIEW

File: packages/rgbpp/src/rgbpp/utils/transaction.ts [hunk 1]
```diff
@@ -22,25 +17,20 @@ export async function sendRgbppTxGroups(props: {
...
+        btc_txid: txid,
         ckb_virtual_result: group.ckbVirtualTxResult,
       });
+      results.push({ btcTxId: txid });
     } catch (e) {
-      result.ckb.error = e;
+      if (e instanceof BtcAssetsApiError) {
+        results.push({ error: e.message });
+      } else {
+        results.push({ error: 'Sending the RGB++ group transactions failed' });
```
Comment (ShookLyngs) [inline packages/rgbpp/src/rgbpp/utils/transaction.ts:31]: Maybe add the `e` to the `error` when it's not BtcAssetsApiError?

File: packages/rgbpp/src/rgbpp/utils/transaction.ts [hunk 2]
```diff
@@ -1,19 +1,14 @@
...
 export interface SentRgbppTxGroup {
-  btc: {
-    txId?: string;
-    error?: Error | unknown;
-  };
-  ckb: {
-    error?: Error | unknown;
-  };
+  btcTxId?: string;
+  error?: string;
```
Comment (ShookLyngs) [inline packages/rgbpp/src/rgbpp/utils/transaction.ts:11]: The `string` type error is inconvenient when tracing the error.

Comment (duanyytop) [reply packages/rgbpp/src/rgbpp/utils/transaction.ts:11]: The python SDK can not parse the TypeScript Error object and  I think the string is enough.
```
[{'error': 'BtcAssetsAPI returned an error: (400) sendrawtransaction RPC error: {"code":-25,"message":"bad-txns-inputs-missingorspent"}'}]
```

File: packages/rgbpp/src/rgbpp/utils/transaction.ts (hunk 1 above)
Comment (duanyytop) [reply packages/rgbpp/src/rgbpp/utils/transaction.ts:31]: The try-catch block only includes btcService RPC, so I don't think another error will occur.

File: packages/rgbpp/src/rgbpp/utils/transaction.ts (hunk 2 above)
Comment (ShookLyngs) [reply packages/rgbpp/src/rgbpp/utils/transaction.ts:11]: > The python SDK can not parse the TypeScript Error object and I think the string is enough.

But non-python users can use this method for their node/browser apps as well.

File: packages/rgbpp/src/rgbpp/utils/transaction.ts (hunk 1 above)
Comment (ShookLyngs) [reply packages/rgbpp/src/rgbpp/utils/transaction.ts:31]: You reminded me that the BtcAssetsApi request could throw a `TypeError` when the `fetch()` fails. I'm not sure if we should wrap the `fetch()` in a try-catch block. The current implementation only catches errors during response parsing. See: [GitHub link](https://github.com/ckb-cell/rgbpp-sdk/blob/develop/packages/service/src/service/base.ts#L38-L57).

```
TypeError: fetch failed
    at node:internal/deps/undici/undici:13737:13
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at BtcAssetsApi.request (/Users/root/Development/projects/rgbpp-sdk/packages/service/src/service/base.ts:38:17)
```

File: packages/rgbpp/src/rgbpp/utils/transaction.ts (hunk 2 above)
Comment (duanyytop) [reply packages/rgbpp/src/rgbpp/utils/transaction.ts:11]: I suggest to add `console.error` to try-catch block for debug

File: packages/rgbpp/src/rgbpp/utils/transaction.ts (hunk 1 above)
Comment (duanyytop) [reply packages/rgbpp/src/rgbpp/utils/transaction.ts:31]: I think the above error message of the `fetch` function is not helpful for developers

Comment (duanyytop) [reply packages/rgbpp/src/rgbpp/utils/transaction.ts:32]: https://github.com/ckb-cell/rgbpp-sdk/pull/283/commits/1c11426f6c1ca6ab196929b0e6641ccfdbdc4c20

**Review by ShookLyngs — DISMISSED** [review]

File: packages/rgbpp/src/rgbpp/utils/transaction.ts (hunk 2 above)
Comment (duanyytop) [reply packages/rgbpp/src/rgbpp/utils/transaction.ts:11]: https://github.com/ckb-cell/rgbpp-sdk/pull/283/commits/1c11426f6c1ca6ab196929b0e6641ccfdbdc4c20

**Review by Dawn-githup — DISMISSED** [review]

File: apps/service/src/rgbpp/types.ts [hunk 3]
```diff
@@ -72,12 +71,9 @@ export interface RgbppTransferAllReq {
...
-  transactions: RgbppTransferAllGroupWithStringCkbVtx[];
-}
-
-export interface RgbppTransferAllGroupWithStringCkbVtx extends Omit<RgbppTransferAllTxGroup, 'ckb'> {
-  ckb: Omit<RgbppTransferAllTxGroup['ckb'], 'virtualTxResult'> & {
-    virtualTxResult: string;
-  };
+export interface RgbppTransferAllResp {
+  ckbVirtualTxResult: string;
+  psbtHex: string;
```
Comment (ShookLyngs) [inline apps/service/src/rgbpp/types.ts:76]:
```suggestion
  btcPsbtHex: string;
```

Comment (duanyytop) [reply apps/service/src/rgbpp/types.ts:76]: PSBT typically refers to BTC by default, so not adding the BTC prefix wouldn't confuse.

Comment (ShookLyngs) [reply apps/service/src/rgbpp/types.ts:76]: That's fine with me, but notice there is a `btcPsbtHex` declared in the types: https://github.com/ckb-cell/rgbpp-sdk/blob/develop/apps/service/src/rgbpp/types.ts#L18-L23

**Review by ShookLyngs — APPROVED** [review]

**Review by Flouse — APPROVED** [review]

**Pull request merged**